java/ql/test/library-tests/frameworks/spring/util/Test.java

```diff
@@ -0,0 +1,1445 @@
+			sink(out); // $hasTaintFlow
+		}
+		{
+			// "org.springframework.util;FileCopyUtils;false;copy;;;Argument[0];Argument[1];taint"
+			File out = null;
+			File in = (File)source();
+			FileCopyUtils.copy(in, out);
+			sink(out); // $hasTaintFlow
+		}
+		{
+			// "org.springframework.util;FileCopyUtils;false;copyToByteArray;;;Argument[0];ReturnValue;taint"
+			byte[] out = null;
+			InputStream in = (InputStream)source();
+			out = FileCopyUtils.copyToByteArray(in);
+			sink(out); // $hasTaintFlow
+		}
+		{
+			// "org.springframework.util;FileCopyUtils;false;copyToByteArray;;;Argument[0];ReturnValue;taint"
+			byte[] out = null;
+			File in = (File)source();
+			out = FileCopyUtils.copyToByteArray(in);
+			sink(out); // $hasTaintFlow
+		}
+		{
+			// "org.springframework.util;FileCopyUtils;false;copyToString;;;Argument[0];ReturnValue;taint"
+			String out = null;
+			Reader in = (Reader)source();
+			out = FileCopyUtils.copyToString(in);
+			sink(out); // $hasTaintFlow
+		}
+		{
+			// "org.springframework.util;FileSystemUtils;false;copyRecursively;(java.io.File,java.io.File);;Argument[0];Argument[1];taint"
+			File out = null;
+			File in = (File)source();
+			FileSystemUtils.copyRecursively(in, out);
+			sink(out); // $hasTaintFlow
+		}
+		{
+			// "org.springframework.util;LinkedMultiValueMap;false;LinkedMultiValueMap;(java.util.Map);;Element of MapValue of Argument[0];Element of MapValue of Argument[-1];value"
+			LinkedMultiValueMap out = null;
+			Map in = Map.of(null, List.of(source()));
+			out = new LinkedMultiValueMap(in);
+			sink(getElement((List)getMapValue(out))); // $hasValueFlow
+		}
+		{
+			// "org.springframework.util;LinkedMultiValueMap;false;LinkedMultiValueMap;(java.util.Map);;MapKey of Argument[0];MapKey of Argument[-1];value"
+			LinkedMultiValueMap out = null;
+			Map in = Map.of(source(), null);
+			out = new LinkedMultiValueMap(in);
+			sink(getMapKey(out)); // $hasValueFlow
+		}
+		{
+			// "org.springframework.util;LinkedMultiValueMap;false;deepCopy;;;MapKey of Argument[-1];MapValue of ReturnValue;value"
+			LinkedMultiValueMap out = null;
+			LinkedMultiValueMap in = (LinkedMultiValueMap)Map.of(source(), null);
+			out = in.deepCopy();
+			sink(getMapValue(out)); // $hasValueFlow
+		}
+		{
+			// "org.springframework.util;LinkedMultiValueMap;false;deepCopy;;;MapValue of Argument[-1];MapValue of ReturnValue;value"
+			LinkedMultiValueMap out = null;
+			LinkedMultiValueMap in = (LinkedMultiValueMap)Map.of(null, source());
+			out = in.deepCopy();
+			sink(getMapValue(out)); // $hasValueFlow
+		}
+		{
+			// "org.springframework.util;MultiValueMap;true;add;;;Argument[0];MapKey of Argument[-1];value"
+			MultiValueMapAdapter out = null;
+			Object in = (Object)source();
+			out.add(in, null);
+			sink(getMapKey(out)); // $hasValueFlow
+		}
+		{
+			// "org.springframework.util;MultiValueMap;true;add;;;Argument[0];MapKey of Argument[-1];value"
+			MultiValueMap out = null;
+			Object in = (Object)source();
```

Comment:
Casting to `Object` is superfluous.
```suggestion
			Object in = source();
```

Reply:
(More instances of this below)